docs(button): add danger button description (#1223)
siemens/ix: packages/documentation/docs/controls/buttons/_button_styleguide.md

## Variants

- **Primary button:** In our applications, we most often use the primary button variant.
- **Secondary button:** The secondary button variant has a grayscale appearance to distinguish it from the primary button.

> - **Danger button:** We use the danger button variant to emphasize particularly dangerous, destructive or critical actions that cannot easily be undone, consider this especially in confirmation dialogs. 

Reviewer: ```suggestion
- **Danger button:** We use the danger button variant to emphasize particularly dangerous, destructive or critical actions that cannot easily be undone,. Consider this especially in confirmation dialogs. 
```

## Behavior in context
- **Interaction:** Buttons can be triggered by pressing anywhere within the button container. When buttons are focused, they can be triggered by pressing `Space`.
- **Text truncation:** Button labels are not truncated. All text on buttons is one line only. 

> - **Ellipsis (...):** We use ellipsis when a button action requires further user input for executing. Example: The button "Connect..." opens a modal with a list of potential connection targets from which the user selects one. This rule does not apply to actions with a subsequent confirmation dialog.

Author: The ellipsis pattern is one of the very classical standards on all usual operating systems used on buttons and menu actions ("Save" vs. "Save as…", "Print" vs. "Print…")

- **Alignment:** Buttons can be left-justified or right-justified or fully span a container’s width.
- **Cluster buttons:** Buttons can be clustered in groups based on their relationship. A cluster can contain a mixture of buttons, e.g. a combination of a primary button with two primary ghost buttons. We recommend a minimum distance of `0.5rem` between adjacent buttons, and we typically cluster buttons if actions/functions are related but not excluding each other. Otherwise, consider using a `button group`. 
- **Button width:** The button width depends on its content. In addition, buttons have a default minimum width of 5rem to layout common combinations such as "OK" and "Cancel" in a harmonic way with equal widths. The minimum width is customizable to accommodate other combinations.

## States

Buttons have six states: Default, hover, active, disabled, loading and focused. In a disabled state, buttons are visually displayed but don’t offer any user interaction.

![Button states](https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?type=design&node-id=132-13020&mode=design&t=KAxDgJoFX436Uk0b-11)

> ## Dos and Don’ts

Reviewer: ```suggestion
## Dos and Don'ts
```

- Do use short button labels to allow users to quickly scan, understand and remember them (follow our writing style guide for more support)
- Do use only one primary (default) button in one visual unit for a clear and singular focus on the main call to action
- Do use the danger button for destructive or critical actions like "delete" or "remove", especially in confirmation dialogs
- Do use a default margin between adjacent buttons of at least `0.5rem`
- Do adjust default and outline button width to label length or container width
- Don’t use the danger button excessively and repetitively as in lists or tables, use it carefully and very selectively instead
- Don’t extend your ghost button width beyond the label’s available width

> - Don’t only rely on standard buttons when many actions/functions are necessary, consider dropdown, split or menu buttons or moving some functionality to a panel or a dialog

Reviewer: ```suggestion
- Don’t only rely on standard buttons when many actions/functions are necessary,. Consider dropdown, split or menu buttons or moving some functionality to a panel or a dialog
```

> - Don’t use the danger button excessively and repetitively as in lists or tables, use it carefully and very selectively instead
> - Don’t extend your ghost button width beyond the label’s available width
> - Don’t only rely on standard buttons when many actions/functions are necessary, consider dropdown, split or menu buttons or moving some functionality to a panel or a dialog

Reviewer: ```suggestion
- Don't use the danger button excessively and repetitively as in lists or tables, use it carefully and very selectively instead
- Don't extend your ghost button width beyond the label’s available width
- Don't only rely on standard buttons when many actions/functions are necessary, consider dropdown, split or menu buttons or moving some functionality to a panel or a dialog
```
